### PR TITLE
Remove entities topics, sub-dimensions, farming systems

### PR DESCRIPTION
**Pull request by dan-tang-ssd**
This PR contains the program change for removing entities topics, sub-dimensions and farming systems.

This includes the following changes:
 - remove related relationships in model classes
 - remove model classes
 - remove relation manager
 - update seeder files
 - update factory files
 - remove database tables topics, sub_dimensions, farming_systems
 - remove column dimensions.topic_id

As there are many changes in many program files, these changes are submitted in a separate PR to minimize the possible impact.

### CODE REVIEW

File: app/Filament/Resources/MetricResource/RelationManagers/DimensionsRelationManager.php [hunk 1]
```diff
@@ -66,18 +56,19 @@ public function table(Table $table): Table
...
-                                ->label('Enter the name of the new dimension'),
-                            Textarea::make('notes')
-                                ->label('Add any notes about this dimension'),
-                        ])
+                    ->recordSelect(
+                        fn (Select $select) => $select
+                            ->multiple()
+                            ->createOptionForm([
+                                // Select::make('topic_id')
+                                //     ->relationship('topic', 'name'),
```
Comment (dave-mills) [inline app/Filament/Resources/MetricResource/RelationManagers/DimensionsRelationManager.php]: May as well delete this entirely!

File: database/factories/DimensionFactory.php [hunk 2]
```diff
@@ -17,7 +17,7 @@ public function definition(): array
             'created_at' => Carbon::now(),
             'updated_at' => Carbon::now(),
 
-            'topic_id' => Topic::factory(),
+            // 'topic_id' => Topic::factory(),
```
Comment (dave-mills) [inline database/factories/DimensionFactory.php]: Likewise here - not that we're ever going to use these factories or seeders again though...

File: database/migrations/2024_03_07_160308_remove_three_tables.php [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        Schema::disableForeignKeyConstraints();
+
+        Schema::table('dimensions', function (Blueprint $table) {
+            $table->dropForeign('dimensions_topic_id_foreign');
+            $table->dropColumn('topic_id');
+        });
+
+        Schema::dropIfExists('topics');
+        Schema::dropIfExists('sub_dimensions');
+        Schema::dropIfExists('farming_systems');
```
Comment (dave-mills) [inline database/migrations/2024_03_07_160308_remove_three_tables.php:27]: Could also remove the link tables (metric_topic, metric_sub_dimension, metric_farming_system)

**Review by dave-mills — APPROVED** [review]
👍  Looks good to me. Good call putting it in separate PR so it's easier to review - thanks!

Couple of extra minor pieces we can remove; then it's good to go.

File: app/Filament/Resources/MetricResource/RelationManagers/DimensionsRelationManager.php (hunk 1 above)
Comment (dan-tang-ssd) [reply app/Filament/Resources/MetricResource/RelationManagers/DimensionsRelationManager.php]: Yes, Thank you Dave.

File: database/factories/DimensionFactory.php (hunk 2 above)
Comment (dan-tang-ssd) [reply database/factories/DimensionFactory.php]: Yes.

File: database/migrations/2024_03_07_160308_remove_three_tables.php (hunk 3 above)
Comment (dan-tang-ssd) [reply database/migrations/2024_03_07_160308_remove_three_tables.php:27]: Good spot!

**Comment by dan-tang-ssd** [comment]
I have removed unnecessary code as advised. 
I will merge this PR now.

**Pull request merged**